src/archive/archive.did

```diff
@@ -0,0 +1,143 @@
+type HeaderField = record { text; text; };
+
+type HttpRequest = record {
+  method: text;
+  url: text;
+  headers: vec HeaderField;
+  body: blob;
+};
+
+type HttpResponse = record {
+  status_code: nat16;
+  headers: vec HeaderField;
+  body: blob;
+  upgrade : opt bool;
+  streaming_strategy: opt StreamingStrategy;
+};
+
+type StreamingCallbackHttpResponse = record {
+  body: blob;
+  token: opt Token;
+};
+
+type Token = record {};
+
+type StreamingStrategy = variant {
+  Callback: record {
+    callback: func (Token) -> (StreamingCallbackHttpResponse) query;
+    token: Token;
+  };
+};
+
+type Timestamp = nat64;
+type UserNumber = nat64;
+type CredentialId = blob;
+type PublicKey = blob;
+type DeviceKey = PublicKey;
+
+type Purpose = variant {
+    recovery;
+    authentication;
+};
+
+type KeyType = variant {
+    unknown;
+    platform;
+    cross_platform;
+    seed_phrase;
+};
+
+type DeviceProtection = variant {
+    protected;
+    unprotected;
+};
+
+type OperationType = variant {
+    register_anchor: record {
+        initial_device: DeviceDataWithoutAlias;
+    };
+    add_device: record {
+        new_device: DeviceDataWithoutAlias;
+    };
+    update_device: record {
+        updated_device: PublicKey;
+        changed_data: DeviceDataUpdate;
+    };
+    remove_device: record {
+        removed_device: PublicKey;
+    };
+};
+
+type LogEntry = record {
+    user_number: UserNumber;
+    operation: OperationType;
+    timestamp: Timestamp;
+    caller: PublicKey;
+    sequence_number: nat64;
+};
+
+type DeviceDataWithoutAlias = record {
+    pubkey: DeviceKey;
+    credential_id: opt  CredentialId;
+    purpose: Purpose;
+    key_type: KeyType;
+    protection: DeviceProtection;
+};
+
+type DeviceDataUpdate = record {
+    alias: opt Hidden;
+    credential_id: opt CredentialId;
+    purpose: opt Purpose;
+    key_type: opt KeyType;
+    protection: opt DeviceProtection;
+};
+
+// Placeholder for information that has been hidden for privacy reasons.
+type Hidden = variant {
+    hidden_for_privacy_reasons;
+};
+
+type Cursor = variant {
+    timestamp: Timestamp; // get logs starting from this Timestamp
+    next_token: blob; // submit previously received token to fetch the next page of log entries
+};
+
+type UserLogs = record {
+  entries: vec blob;
+  next_token: opt blob // cursor to fetch the next page of log entries (if any)
+};
+
+type Logs = record {
+  entries: vec blob;
+  next_idx: opt nat64 // next index to fetch the next page of log entries (if any)
+};
+
+type ArchiveInit = record {
+  ii_canister: principal;
```

Comment:
how about not actually mentioning II here or in the `write_entry` doc below?

I'm thinking like this: someone who sees `ii_canister` may assume that somehow the archive canister needs to authenticate or call the II canister to function; if the attribute name was e.g. `owner` or `writer` or something more descriptive, then it would be clearer

Reply:
👍